src/components/Request/NewRequestForm.Wizard.tsx

```diff
@@ -0,0 +1,193 @@
+    return usaJobsFlyer
+      ? "USAJobs"
+      : INPUTSTEPS.get("USAJobs")?.next(methods) ?? "USAJobs";
+  },
+  prev: (methods: string[]) => {
+    const linkedinPost = methods.includes("linkedinPost");
+    return linkedinPost
+      ? "LinkedInPost"
+      : INPUTSTEPS.get("LinkedInPost")?.prev(methods) ?? "LinkedInPost";
+  },
+});
+
+INPUTSTEPS.set("USAJobs", {
+  next: (_methods: string[]) => "DONE",
+  prev: (methods: string[]) => {
+    const linkedinSearch = methods.includes("linkedinSearch");
+    return linkedinSearch
+      ? "LinkedInSearch"
+      : INPUTSTEPS.get("LinkedInSearch")?.prev(methods) ?? "LinkedInSearch";
+  },
+});
+
+INPUTSTEPS.set("DONE", {
+  next: (_methods: string[]) => "DONE",
+  prev: (methods: string[]) => {
+    const linkedinSearch = methods.includes("usaJobsFlyer");
+    return linkedinSearch
+      ? "USAJobs"
+      : INPUTSTEPS.get("USAJobs")?.prev(methods) ?? "USAJobs";
+  },
+});
+
+interface ActionType {
+  type: string;
+  payload?: string[];
+}
+
+function reducer(state: { page: string }, action: ActionType) {
+  switch (action.type) {
+    case "next_page":
+      const nextPage = INPUTSTEPS.get(state.page)?.next(action.payload ?? []);
+
+      return nextPage ? { ...state, page: nextPage } : state;
+
+    case "prev_page":
+      const prevPage = INPUTSTEPS.get(state.page)?.prev(action.payload ?? []);
+      return prevPage ? { ...state, page: prevPage } : state;
+
+    case "reset":
+      return { page: "RoutingInfo" };
+
+    case "goto":
+      const step = action.payload?.[0];
+      return step ? { ...state, page: step } : state;
+
+    default:
+      return state;
+  }
+}
+
+const Wizard = ({ isLoading = false, isError = false }) => {
+  const [state, dispatch] = useReducer(reducer, { page: "RoutingInfo" });
+  const form = useFormContext();
+  const methods = form.watch("methods");
+  const gotoStep = (step: string) => {
+    const steps: string[] = [];
+    steps.push(step);
+    dispatch({ type: "goto", payload: steps });
+  };
+  return (
+    <>
+      {state.page === "RoutingInfo" && <RoutingInfo />}
+      {state.page === "HiringInfo" && <HiringInfo />}
+      {state.page === "JobBoard" && <JobBoard />}
+      {state.page === "JOA" && <JOA />}
+      {state.page === "LinkedInPost" && <LinkedInPost />}
+      {state.page === "LinkedInSearch" && <LinkedInSearch />}
+      {state.page === "USAJobs" && <USAJobs />}
+      {state.page === "DONE" && <Done gotoStep={gotoStep} />}
+
+      {!isLoading && (
+        <div className="requestWizardButtons">
+          <Button
+            disabled={state.page === "RoutingInfo" || isLoading}
+            onClick={() => dispatch({ type: "prev_page", payload: methods })}
+          >
+            Prev
+          </Button>
+          <Button
+            style={{ marginLeft: "auto" }}
+            disabled={state.page === "DONE" || isLoading}
+            onClick={() => {
+              form.trigger();
+              if (form.formState.isValid) {
```

Comment:
You should do something similar on the Prev page button.  This will prevent users from entering bad data and it being stored (or erroring on attempting to store).  Example if you fill a text field with more than allowed characters, and hit "Prev" then "Save" -- it will error in the HTTP call since it exceeds SP field.